runtime/browser/runtime.cc

```diff
@@ -316,6 +317,16 @@ void Runtime::TitleWasSet(content::NavigationEntry* entry, bool explicit_set) {
     ui_delegate_->UpdateTitle(entry->GetTitle());
 }
 
+void Runtime::DidNavigateAnyFrame(
+    content::RenderFrameHost* render_frame_host,
+    const content::LoadCommittedDetails& details,
+    const content::FrameNavigateParams& params) {
+  DCHECK(thread_checker_.CalledOnValidThread());
+
+  XWalkBrowserContext::FromWebContents(web_contents())
```

Comment:
I think it's better like it is, otherwise your suggestion needs a cast from a BrowserContext to a XWalkBrowserContext.